idseq_dag/steps/download_accessions.py

```diff
@@ -44,13 +45,71 @@ def run(self):
+        '''
+            We found a ncbi record that is oddly annotated (title starts with a comma,
+            as you can see here: https://www.ncbi.nlm.nih.gov/protein/XP_002289390.1)
+            That produce wrong results in blastx (blastx understand this accession
+            as being "XP_002289390.1," instead of "XP_002289390.1")
+
+            This method detects and removes this trailling comma from the title, and
+            it is being invoked before writing it to file assembly/nr.refseq.fasta,
+            which is a subset of the original nr index.
+
+            `accession_data` is a string that looks like this:
+
+                accession_data = (
+                    '>XP_002289390.1 , partial [Thalassiosira pseudonana CCMP1335]\x01EED92927.1 Conserved Hypothetical Protein, partial [Thalassiosira pseudonana CCMP1335]\n'
+                    'GGREKKKLLKSQKDGSAKDRHNPRAFSVANIVRTQRNVQRNLDRAQKKEYVPLSDRRAARVEEGPPSLVAVVGPPGVGKS\n'
+                    'TLIRSLVKLYTNHNLTNPTGPITVCTSQTKRITFLECPNTPTAMLDVAKIADLVLLCVDAKFGFEMETFEFLNMMQTHGF\n'
+                    'PKVMGIFTHLDQFRTQKNLRKTKKLLKHRFWTEIYDGAKMFYFSGCVNGKYLKHEVKQLSLLLSRIKYRPLVWRNTHPYV\n'
+                    # (...more lines with sequence data...)
+                )
+                # note entry above represents a single string and line breaks and ^A character are part of it.
+
+            this method returns same string with title correction:
+
+            result = (
+                '>XP_002289390.1 partial [Thalassiosira pseudonana CCMP1335]\x01EED92927.1 Conserved Hypothetical Protein, partial [Thalassiosira pseudonana CCMP1335]\n'
+                'GGREKKKLLKSQKDGSAKDRHNPRAFSVANIVRTQRNVQRNLDRAQKKEYVPLSDRRAARVEEGPPSLVAVVGPPGVGKS\n'
+                'TLIRSLVKLYTNHNLTNPTGPITVCTSQTKRITFLECPNTPTAMLDVAKIADLVLLCVDAKFGFEMETFEFLNMMQTHGF\n'
+                'PKVMGIFTHLDQFRTQKNLRKTKKLLKHRFWTEIYDGAKMFYFSGCVNGKYLKHEVKQLSLLLSRIKYRPLVWRNTHPYV\n'
+                # (...more lines with sequence data...)
+            )
+
+            Right now this method is only being used to remove the trailing comma,
+            since this is the only case that we found so far affecting the pipeline.
+            It may be extended the future to handle more exceptions if it is needed.
+        '''
+        def _fix_headers(line):
+            if len(line) > 0 and line[0] == ">":
+                # support for multiheader line separted by CTRL_A (https://en.wikipedia.org/wiki/FASTA_format#Description_line)
+                header_items = line.lstrip(">").split("\x01")
+                return ">" + "\x01".join(map(_fix_comma, header_items))
+            return line
+
+        def _fix_comma(header_item):
+            sub_items = header_item.split(" ", 1)
+            if len(sub_items) > 1:
+                accession_id, description = sub_items
+                if len(description) > 0 and description[0] == ",":
+                    description = description.lstrip(",").lstrip(" ")
```

Comment:
This seems too convoluted to remove the first comma after an accession id. What is the catch?

Reply:
It is removing the comma before the description. The problem is that `blastx` mistakenly interprets that as being part of the accession id, even when in fact there is a space between these things.

Reply:
Yes, but is just the first comma after the accession id, right? At first glance seems like a regex based sub could take care of this...

Reply:
New version is using a regular expression with named group captures. Might be easier to grasp now.